metals/src/main/scala/scala/meta/internal/search/SymbolHierarchyOps.scala

```diff
@@ -0,0 +1,227 @@
+package scala.meta.internal.search
+
+import scala.util.control.NonFatal
+
+import scala.meta.internal.implementation.MethodImplementation
+import scala.meta.internal.implementation.TextDocumentWithPath
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ScalaVersionSelector
+import scala.meta.internal.mtags.GlobalSymbolIndex
+import scala.meta.internal.mtags.Mtags
+import scala.meta.internal.mtags.Semanticdbs
+import scala.meta.internal.mtags.SymbolDefinition
+import scala.meta.internal.mtags.{Symbol => MSymbol}
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.search.SymbolHierarchyOps._
+import scala.meta.internal.semanticdb.ClassSignature
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.TypeRef
+import scala.meta.internal.symtab.GlobalSymbolTable
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.Location
+
+class SymbolHierarchyOps(
```

Comment:
All those things are moved from `ImplementationProvider`. Those are not needed for go to implementation anymore but are still used by `Supermethods` and `RenameProvider`. I didn't want to try and remove those in this PR.